### PR TITLE
Add Hawaii income tax calculation logic

### LINKED ISSUE #3274: Add Hawaii income tax calculation logic
The formula for the `hi_income_tax` variable is missing even though `hi_income_tax_before_credits` is being calculated along with several Hawaii credits.

Adding this missing formula would allow the testing against TAXSIM35 to begin.

### PR DESCRIPTION
**Pull request by YichunRen**
Fixes #3274

### Summary
🌴💵🧮

This pull request adds the logic for calculating Hawaii income tax as a minor feature. This is a step towards implementing state-level tax policies in `policyengine-us`.

> _`changelog` updated_
> _Hawaii income tax logic_
> _a minor feature_

### Walkthrough
*  Add logic for calculating Hawaii income tax ([link](https://github.com/PolicyEngine/policyengine-us/pull/3290/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4),                              F30

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ - bump: minor
+  changes:
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:2]: indentation

File: policyengine_us/parameters/gov/states/hi/tax/income/credits/non_refundable.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+description: Hawaii non-refundable tax credits.
+values:
+  2022-01-01:
+    - hi_cdcc
+    - hi_eitc
+    - hi_tax_credit_for_low_income_household_renters
+metadata:
+  unit: list
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/hi/tax/income/credits/non_refundable.yaml:8]:
```suggestion
  unit: list
  period: year
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/hi/tax/income/credits/non_refundable.yaml:8]: all yaml files

File: policyengine_us/parameters/gov/states/hi/tax/income/credits/non_refundable.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  label: Hawaii non-refundable tax credits
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/hi/tax/income/credits/non_refundable.yaml:9]: add one tax form reference 2021 + 2022

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/hi/tax/income/credits/non_refundable.yaml:9]: all yaml

File: policyengine_us/variables/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.py:1]: rename `hi_income_tax_before_credits` to `hi_income_tax_before_non_refundable_credits`

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.py:1]: same for file name and test file name

File: policyengine_us/variables/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from policyengine_us.model_api import *
+
+
+class hi_income_tax_before_refundable_credits(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Hawaii income tax"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.py:7]:
```suggestion
    label = "Hawaii income tax before refundable credits"
```

File: policyengine_us/parameters/gov/states/hi/tax/income/credits/refundable.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+description: Hawaii refundable tax credits.
+values:
+  2022-01-01:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/hi/tax/income/credits/refundable.yaml:3]:
```suggestion
  2021-01-01:
```

File: policyengine_us/parameters/gov/states/hi/tax/income/credits/non_refundable.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+description: Hawaii non-refundable tax credits.
+values:
+  2022-01-01:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/hi/tax/income/credits/non_refundable.yaml:3]:
```suggestion
  2021-01-01:
```

File: policyengine_us/variables/gov/states/hi/tax/income/hi_non_refundable_credits.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+from policyengine_us.model_api import *
+
+
+class hi_non_refundable_credits(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Hwaii non-refundable tax credits"
```
Comment (YichunRen) [inline policyengine_us/variables/gov/states/hi/tax/income/hi_non_refundable_credits.py:7]: Hawaii

File: changelog_entry.yaml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Add Hawaii income tax calculation logic.
```
Comment (LeqiGogo) [inline changelog_entry.yaml:4]:
```suggestion
    - Hawaii income tax calculation logic.
```

File: policyengine_us/parameters/gov/states/hi/tax/income/credits/refundable.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    - hi_tax_credit_for_low_income_household_renters
+metadata:
+  unit: list
+  period: year
+  label: Hawaii refundable tax credits
+  reference:
+    - title: 2022 Hawaii N-11 Tax Form
+      href: https://files.hawaii.gov/tax/forms/2022/n11_i.pdf#page=2
+    - title: 2021 Hawaii N-11 Tax Form
+      href: https://files.hawaii.gov/tax/forms/2021/n11_i.pdf#page=3
```
Comment (LeqiGogo) [inline policyengine_us/parameters/gov/states/hi/tax/income/credits/refundable.yaml:16]: add legal code

File: policyengine_us/tests/policy/baseline/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.yaml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+- name: 100 hi_income_tax_before_credits
+  period: 2022
+  input:
+    hi_income_tax_before_credits: 100
+    state_code: HI
+  output:
+    hi_income_tax_before_refundable_credits: 100
+
```
Comment (LeqiGogo) [inline policyengine_us/tests/policy/baseline/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.yaml:8]: need test for all variables
need more test cases

File: policyengine_us/variables/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+class hi_income_tax_before_refundable_credits(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Hawaii income tax before refundable credits"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.HI
+
+    def formula(tax_unit, period, parameters):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.py:12]: use an adds function here

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.py:12]: adds = ["hi_refundable_credits"]

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.py:12]: leave the commented part

File: policyengine_us/tests/policy/baseline/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/hi/tax/income/hi_income_tax_before_refundable_credits.yaml:1]: dont need this then

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**